src/Test/Perf/util/runner_util.csx

```diff
@@ -38,3 +40,134 @@ async Task<ScriptState<object>> RunFile(string fileName)
+                    {
+                        var metricKey = xmlReader.GetAttribute("Name");
+
+                        if (string.Equals(metricKey, requiredMetricKey))
+                        {
+                            var metricScale = xmlReader.GetAttribute("Units");
+                            xmlReader.Read();
+                            var metricvalue = xmlReader.Value;
+                            result.Add($"{currentScenarioName}, {metricKey} ({metricScale}), {metricvalue}");
+                        }
+                    }
+                }
+            }
+        }
+
+        File.WriteAllLines(destination, result);
+    }
+    catch(System.Exception e)
+    {
+        System.Console.WriteLine(e.Message);
+        System.Console.WriteLine(e.StackTrace);
+        return false;
+    }
+
+    return true;
+}
+
+/// Gets a csv file with metrics and converts them to ViBench supported JSON file
+string GetViBenchJsonFromCsv(string compilerTimeCsvFilePath, string execTimeCsvFilePath, string fileSizeCsvFilePath)
+{
+    string branch = StdoutFrom("git", "rev-parse --abbrev-ref HEAD");
+    string date = FirstLine(StdoutFrom("git", $"show --format=\"%aI\" {branch} --"));
+    string hash = FirstLine(StdoutFrom("git", $"show --format=\"%h\" {branch} --"));
+    string longHash = FirstLine(StdoutFrom("git", $"show --format=\"%H\" {branch} --"));
+    string username = StdoutFrom("whoami");
+    string machineName = StdoutFrom("hostname");
+    string architecture = System.Environment.Is64BitOperatingSystem ? "x86-64" : "x86";
+
+    // File locations
+    string outJson = Path.Combine(GetCPCDirectoryPath(), $"Roslyn-{longHash}.json");
+
+    // ViBenchToJson does not like empty csv files.
+    string files = "";
+    if (compilerTimeCsvFilePath != null && new FileInfo(compilerTimeCsvFilePath).Length != 0) {
+        files += $@"compilertime:""{compilerTimeCsvFilePath}""";
+    }
+    if (execTimeCsvFilePath != null && new FileInfo(execTimeCsvFilePath).Length != 0) {
+        files += $@"exectime:""{execTimeCsvFilePath}""";
+    }
+    if (fileSizeCsvFilePath != null && new FileInfo(fileSizeCsvFilePath).Length != 0) {
+        files += $@"filesize:""{fileSizeCsvFilePath}""";
+    }
+    string arguments = $@"
+    {files}
+    jobName:""RoslynPerf-{hash}-{date}""
+    jobGroupName:""Roslyn-{branch}""
+    jobTypeName:""official""
+    buildInfoName:""{date}-{branch}-{hash}""
+    configName:""Default Configuration""
+    machinePoolName:""4-core-windows""
+    architectureName:""{architecture}""
+    manufacturerName:""unknown-manufacturer""
+    microarchName:""unknown-microarch""
+    userName:""{username}""
+    userAlias:""{username}""
+    osInfoName:""Windows""
+    machineName:""{machineName}""
+    buildNumber:""{date}-{hash}""
+    /json:""{outJson}""
+    ";
+
+    arguments = arguments.Replace("\r\n", " ").Replace("\n", "");
+
+    ShellOutVital(@"\\mlangfs1\public\basoundr\vibenchcsv2json\ViBenchToJson.exe", arguments);
```

Comment:
I would like to keep bootstrap.bat as small as possible and move this to `automation.csx` will download the tools that are necessary for the automation run